src/openloco/windows/toolbar_top.cpp

```diff
@@ -0,0 +1,1236 @@
+
+        if (current_viewport_flags & viewport_flags::height_marks_on_tracks_roads)
+            dropdown::set_item_selected(4);
+
+        if (current_viewport_flags & viewport_flags::height_marks_on_land)
+            dropdown::set_item_selected(5);
+
+        if (current_viewport_flags & viewport_flags::one_way_direction_arrows)
+            dropdown::set_item_selected(6);
+
+        if (!(current_viewport_flags & viewport_flags::town_names_displayed))
+            dropdown::set_item_selected(8);
+
+        if (!(current_viewport_flags & viewport_flags::station_names_displayed))
+            dropdown::set_item_selected(9);
+
+        dropdown::set_highlighted_item(0);
+    }
+
+    // 0x0043AF37
+    static void view_menu_dropdown(window* window, widget_index widgetIndex, int16_t itemIndex)
+    {
+        if (itemIndex == -1)
+            itemIndex = dropdown::get_highlighted_item();
+
+        window = WindowManager::getMainWindow();
+        auto viewport = WindowManager::getMainWindow()->viewports[0];
+
+        if (itemIndex == 0)
+            viewport->flags ^= viewport_flags::underground_view;
+        else if (itemIndex == 1)
+            viewport->flags ^= viewport_flags::hide_foreground_tracks_roads;
+        else if (itemIndex == 2)
+            viewport->flags ^= viewport_flags::hide_foreground_scenery_buildings;
+        else if (itemIndex == 4)
+            viewport->flags ^= viewport_flags::height_marks_on_tracks_roads;
+        else if (itemIndex == 5)
+            viewport->flags ^= viewport_flags::height_marks_on_land;
+        else if (itemIndex == 6)
+            viewport->flags ^= viewport_flags::one_way_direction_arrows;
+        else if (itemIndex == 8)
+            viewport->flags ^= viewport_flags::town_names_displayed;
+        else if (itemIndex == 9)
+            viewport->flags ^= viewport_flags::station_names_displayed;
+
+        window->invalidate();
+    }
+
+    // 0x0043A3C3
+    static void terraform_menu_mouse_down(window* window, widget_index widgetIndex)
+    {
+        auto interface = objectmgr::get<interface_skin_object>();
+        auto land = objectmgr::get<land_object>(addr<0x00525FB6, uint8_t>());
+        auto water = objectmgr::get<water_object>();
+
+        dropdown::add(0, string_ids::menu_sprite_stringid, { interface->img + interface_skin::image_ids::toolbar_menu_bulldozer, string_ids::menu_clear_area });
+        dropdown::add(1, string_ids::menu_sprite_stringid, { land->var_16 + 3, string_ids::menu_adjust_land });
+        dropdown::add(2, string_ids::menu_sprite_stringid, { water->var_06 + 42, string_ids::menu_adjust_water });
+        dropdown::add(3, string_ids::menu_sprite_stringid, { interface->img + interface_skin::image_ids::toolbar_menu_plant_trees, string_ids::menu_plant_trees });
+        dropdown::add(4, string_ids::menu_sprite_stringid, { interface->img + interface_skin::image_ids::toolbar_menu_build_walls, string_ids::menu_build_walls });
```

Comment:
This item is supposed to be hidden if `screen_flags & 2` (the editor toolbar uses the same code for populating its menus)

Reply:
I've deliberately omitted this condition, as making the wall tab available in-game was a feature request. Previously, this was implemented by replacing the condition with a `write_nop` hook.